tests/v2/validation/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,450 @@
+	}
+	for _, expectedDeploymentName := range deploymentNames {
+		doesContainDeployment := containsItemWithPrefix(deploymentList.Names(), expectedDeploymentName)
+		assert.Truef(u.T(), doesContainDeployment, "Deployment with prefix %s doesn't exists", expectedDeploymentName)
+	}
+
+	u.T().Logf("Checking daemonsets in namespace %s", u.namespace.Name)
+	daemonsetList, err := daemonsets.ListDaemonsets(client, u.project.ClusterID, u.namespace.Name, defaultListOptions)
+	require.NoError(u.T(), err)
+	daemonsetNames := []string{
+		u.names.coreWithSuffix["daemonsetName"],
+		u.names.coreWithSuffix["daemonsetNameForIngress"],
+	}
+	for _, expectedDaemonsetName := range daemonsetNames {
+		doesContainDaemonset := containsItemWithPrefix(daemonsetList.Names(), expectedDaemonsetName)
+		assert.Truef(u.T(), doesContainDaemonset, "Daemonset with prefix %s doesn't exists", expectedDaemonsetName)
+	}
+
+	if client.Flags.GetValue(environmentflag.Ingress) {
+		u.T().Logf("Ingress tests are enabled")
+
+		u.T().Logf("Checking ingresses in namespace %s", u.namespace.Name)
+		ingressList, err := ingresses.ListIngresses(client, u.project.ClusterID, u.namespace.Name, defaultListOptions)
+		require.NoError(u.T(), err)
+		ingressNames := []string{
+			u.names.coreWithSuffix["ingressNameForDeployment"],
+			u.names.coreWithSuffix["ingressNameForDaemonset"],
+		}
+		for _, expectedIngressName := range ingressNames {
+			doesContainIngress := containsItemWithPrefix(ingressList.Names(), expectedIngressName)
+			assert.Truef(u.T(), doesContainIngress, "Ingress with prefix %s doesn't exists", expectedIngressName)
+
+			if doesContainIngress {
+				ingressName := getItemWithPrefix(ingressList.Names(), expectedIngressName)
+				ingress, err := ingresses.GetIngressByName(client, u.project.ClusterID, u.namespace.Name, ingressName)
+				require.NoError(u.T(), err)
+
+				u.T().Logf("Checking if the ingress %s is accessible", ingress.Name)
+				isIngressAcessible, err := waitUntilIngressIsAccessible(client, ingress.Spec.Rules[0].Host)
+				require.NoError(u.T(), err)
+				assert.True(u.T(), isIngressAcessible)
+			}
+		}
+	}
+
+	u.T().Logf("Checking the secret in namespace %s", u.namespace.Name)
+	secretList, err := secrets.ListSecrets(client, u.project.ClusterID, u.namespace.Name, defaultListOptions)
+	require.NoError(u.T(), err)
+	doesContainSecret := containsItemWithPrefix(secretList.Names(), u.names.core["secretName"])
+	assert.Truef(u.T(), doesContainSecret, "Secret with prefix %s doesn't exists", u.names.core["secretName"])
+
+	if client.Flags.GetValue(environmentflag.Chart) {
+		u.T().Logf("Chart tests are enabled")
+
+		u.T().Logf("Checking if the logging chart is installed")
+		loggingChart, err := charts.GetChartStatus(client, u.project.ClusterID, charts.RancherLoggingNamespace, charts.RancherLoggingName)
+		require.NoError(u.T(), err)
+		assert.True(u.T(), loggingChart.IsAlreadyInstalled)
+	}
+
+	u.T().Logf("Running the pre-upgrade checks")
+	u.TestPreUpgrade()
```

Comment:
It's probably me not understanding but why are we calling the pre-upgrade from the post-upgrade

Reply:
The pre-upgrade test creates the resources with randomly generated names and validates the creations and additional steps. Since all the names are randomly generated, this gives the advantage of running those steps again after the post-upgrade checks. 

This step is to call the pre-upgrade test again to rerun the pre-upgrade test steps.